### PR TITLE
Add guard against zero masked tokens in loss reduction class.

### PR DESCRIPTION
**Pull request by skothenhill-nv**
### Description
If there happens to be no masked tokens in a given microbatch, it replaces it with `1` to avoid throwing a NaN during division.

DO NOT MERGE until the right pieces are cherrypicked.

### Type of changes

- [x]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Refactor
- [ ]  Documentation update
- [ ]  Other (please describe):

### CI Pipeline Configuration
Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

* If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
* If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
TODO: Add code snippet
```

### Pre-submit Checklist

 - [ ] I have tested these changes locally
 - [ ] I have updated the documentation accordingly
 - [ ] I have added/updated tests as needed
 - [ ] All existing tests pass successfully

### CODE REVIEW

File: sub-packages/bionemo-llm/src/bionemo/llm/data/masking.py [hunk 1]
```diff
@@ -36,6 +36,8 @@ class BertMaskConfig:
     mask_prob: float = 0.15
     mask_token_prob: float = 0.8
     random_token_prob: float = 0.1
+    minimum_masked_tokens: int = 1
```
Comment (jstjohn) [inline sub-packages/bionemo-llm/src/bionemo/llm/data/masking.py]: one idea: make this an Optional[int] = None, then if `mask_prob==0` default it to 0 otherwise 1, unless the user supplies something not None?

Comment (jstjohn) [reply sub-packages/bionemo-llm/src/bionemo/llm/data/masking.py]: So like:
```python
if minimum_masked_tokens is None:
    # determine default
    if mask_prob == 0:
        self.minimum_masked_tokens = 0
    else:
        self.minimum_masked_tokens = 1
else:
    self.self.minimum_masked_tokens = minimum_masked_tokens
```

Comment (jstjohn) [reply sub-packages/bionemo-llm/src/bionemo/llm/data/masking.py]: you can do that in a post_init or something

Comment (skothenhill-nv) [reply sub-packages/bionemo-llm/src/bionemo/llm/data/masking.py]: Done!

File: sub-packages/bionemo-llm/src/bionemo/llm/model/loss.py [hunk 2]
```diff
@@ -97,6 +97,10 @@ def forward(
 
         num_valid_tokens = num_valid_tokens.clone().detach().to(torch.int)
         loss_sum_and_ub_size = torch.cat([loss_sum.clone().detach().view(1), num_valid_tokens.view(1)])
+
+        if num_valid_tokens.item() == 0:
+            num_valid_tokens = torch.tensor(1, device=num_valid_tokens.device, dtype=num_valid_tokens.dtype)
```
Comment (skothenhill-nv) [inline sub-packages/bionemo-llm/src/bionemo/llm/model/loss.py]:
```suggestion
            num_valid_tokens = torch.ones_like(num_valid_tokens)
```

File: sub-packages/bionemo-llm/src/bionemo/llm/model/loss.py [hunk 3]
```diff
@@ -97,6 +97,10 @@ def forward(
 
         num_valid_tokens = num_valid_tokens.clone().detach().to(torch.int)
         loss_sum_and_ub_size = torch.cat([loss_sum.clone().detach().view(1), num_valid_tokens.view(1)])
+
```
Comment (jstjohn) [inline sub-packages/bionemo-llm/src/bionemo/llm/model/loss.py]: Comment would be good here.

Comment (jstjohn) [reply sub-packages/bionemo-llm/src/bionemo/llm/model/loss.py]:
```suggestion
# Set to 1 to avoid divide by zero in the megatron scheduler:
#  https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/core/pipeline_parallel/schedules.py#L303-L308
```

File: VERSION [hunk 4]
```diff
@@ -1 +1 @@
-2.6
+2.6.2
```
Comment (pstjohn) [inline VERSION:1]: was this intentional?

**Review by pstjohn — APPROVED** [review]

File: VERSION (hunk 4 above)
Comment (jstjohn) [reply VERSION:1]: Yes, according to @trvachov

File: sub-packages/bionemo-llm/src/bionemo/llm/model/loss.py [hunk 5]
```diff
@@ -97,6 +97,11 @@ def forward(
 
         num_valid_tokens = num_valid_tokens.clone().detach().to(torch.int)
         loss_sum_and_ub_size = torch.cat([loss_sum.clone().detach().view(1), num_valid_tokens.view(1)])
+        # Set to 1 to avoid divide by zero in the megatron scheduler:
+        #  https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/core/pipeline_parallel/schedules.py#L303-L308
+        if num_valid_tokens.item() == 0:
```
Comment (dorotat-nv) [inline sub-packages/bionemo-llm/src/bionemo/llm/model/loss.py:102]: could we use unit tests for that? how do we know it works?
